VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsActivity.java

```diff
@@ -74,6 +74,11 @@ public Fragment getFirstFragment() {
         } else {
             fragment = courseDiscussionPostsThreadFragment;
         }
+        if (fragment.getArguments() == null) {
+            Bundle args = new Bundle();
+            args.putSerializable(Router.EXTRA_ENROLLMENT_COURSE_DATA, courseData);
+            fragment.setArguments(args);
```

Comment:
This should be done generically in `BaseSingleFragmentActivity` I guess, instead of needing to have the Fragments query the Activity's launching Intent extras. This would make the Fragments much more modular, as can be seen here.